kore/src/main/scala/org/kframework/definition/outer-ext.scala

```diff
@@ -104,6 +104,9 @@ case class FlatModule(name: String, imports: Set[Import], localSentences: Set[Se
       throw KEMException.compilerError(msg)
     }
 
+    if (koreModules.contains(mainModName))
+      return koreModules(mainModName)
+
```

Comment:
We should be returning the `FlatModule` associated with `name`, not with `mainModName`.
Shouldn't this read:

```suggestion
    if (koreModules.contains(name))
      return koreModules(name)

```

Reply:
Ah I see above we say that `mainModName = this.name`, but I would still prefer we use `this.name` here, as it expresses the intent of this function better (and thus may behave better for later changes which may introduce or change code here).

Reply:
`var mainModName = this.name`
It's the same thing. I'm going to remove `mainModName` since it's redundant and confusing.